server/database/repositories/TournamentRepository.ts

```diff
@@ -1,37 +1,89 @@
+  };
+  return matchDb;
+}
+function toMatch(matchDoc: MatchDocument) : MatchWithoutMS {
+  const match : MatchWithoutMS = {
+    id: matchDoc._id.toString(),
+    teamA: toTeam(matchDoc.teamA),
+    teamB: toTeam(matchDoc.teamB),
+    score: matchDoc.score,
+    isFinished: matchDoc.isFinished,
+  };
+  return match;
+}
+interface TournamentDb extends Omit<TournamentWihtoutMS, 'id' | 'teams' | 'matches'> {
+  _id?: string,
+  teams: TeamDb[],
+  matches: MatchDb[]
+}
+function toTournamentDb(t: TournamentWihtoutMS) : TournamentDb {
+  const tDb : TournamentDb = {
+    _id: t.id,
+    name: t.name,
+    ownerId: t.ownerId,
+    teams: t.teams.map((team) => toTeamDb(team)),
+    matches: t.matches.map((match) => toMatchDb(match)),
+  };
+  return tDb;
+}
+function toTournament(tourDoc: TournamentDocument) : TournamentWihtoutMS {
+  const tournament : TournamentWihtoutMS = {
+    id: tourDoc._id.toString(),
+    ownerId: tourDoc.ownerId,
+    name: tourDoc.name,
+    teams: tourDoc.teams.map((teamDoc) => toTeam(teamDoc)),
+    matches: tourDoc.matches.map((matchDoc) => toMatch(matchDoc)),
+  };
+  return tournament;
+}
 
 export default class TournamentRepository {
   public static async create(tournament: TournamentWihtoutMS) {
-    // Check if teams have ids, if teams in matches have ids
-    // and if teams in matches are on teams list
+    if (tournament.id) { throw new AppError('New tournament should not have id', 400); }
     tournament.teams.forEach((team) => {
-      if (!team._id) { throw new AppError('All teams should have ids.', 400); }
+      if (!team.id) { throw new AppError('All teams should have ids.', 400); }
     });
     tournament.matches.forEach((match) => {
-      if (!match.teamA._id || !match.teamB._id) { throw new AppError('All teams in matches entries should have ids', 400); }
+      if (!match.teamA.id || !match.teamB.id) { throw new AppError('All teams in matches entries should have ids', 400); }
     });
-
-    return await TournamentModel.create(tournament);
+    const tDoc = await TournamentModel.create(toTournamentDb(tournament));
+    const tour = await TournamentRepository.getById(tDoc._id);
+    if (!tour) { throw new AppError('Error while creating tournament', 400); }
+    return tour;
   }
 
-  public static async updateMatch(matchId: string, match: Match) {
-    if (!matchId) throw new AppError('Provided match does not contain id', 400);
-
-    const tournament = await TournamentModel.findOneAndUpdate(
-      { 'matches._id': matchId },
+  public static async updateMatch(match: MatchWithoutMS) {
+    if (!match.id) throw new AppError('Provided match does not contain id', 400);
+    const matchDb = toMatchDb(match);
+    const tDoc = await TournamentModel.findOneAndUpdate(
+      { 'matches._id': matchDb._id },
       {
         $set: {
-          'matches.$': match,
+          'matches.$': matchDb,
         },
       },
       { new: true },
-    ).lean();
-
-    return tournament;
+    ).exec();
```

Comment:
Why exec()?

Reply:
https://mongoosejs.com/docs/promises.html#should-you-use-exec-with-await?